mlflow/tracking/client.py

```diff
@@ -2394,7 +2394,9 @@ def log_image(
 
             # Sanitize key to use in filename (replace / with # to avoid subdirectories)
             sanitized_key = re.sub(r"/", "#", key)
-            filename_uuid = uuid.uuid4()
+            # Add a prefix to prevent percent encoding when '%' precedes the filename.
+            # For example, if `filename_uuid` starts with '28', it will be interpreted as '('.
+            filename_uuid = "a" + str(uuid.uuid4())[1:]
```

Comment:
I don't think this is a good solution. The same thing occurs for `step` or `timestamp`.